packages/lumx-react/src/components/progress/Progress.tsx

```diff
@@ -12,15 +12,29 @@ import { getRootClassName, handleBasicClasses } from '@lumx/react/utils/classNam
  */
 export const ProgressVariant = { linear: 'linear', circular: 'circular' } as const;
 export type ProgressVariant = ValueOf<typeof ProgressVariant>;
+export type ProgressCircularSize = Extract<Size, 'xxs' | 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl'>;
 
 /**
  * Defines the props of the component.
  */
-export interface ProgressProps extends GenericProps, HasTheme {
+export interface ProgressCircularProps extends GenericProps, HasTheme {
     /** Progress variant. */
-    variant?: ProgressVariant;
+    variant?: 'circular';
+    size?: ProgressCircularSize;
 }
 
+export interface ProgressLinearProps extends GenericProps, HasTheme {
+    /** Progress variant. */
+    variant?: 'linear';
+}
+
+export type ProgressProps =
+    | ProgressCircularProps
+    | (ProgressLinearProps & {
+          /** Progress variant. */
+          variant?: ProgressVariant;
+      });
```

Comment:
We need that:
```suggestion
    variant: 'linear';
}

export type ProgressProps = ProgressCircularProps | ProgressLinearProps;
```
I order to have `<Progress size="foo" />` return an error since the default variant is 'circular' and thus the size must use the correct format